flux-middle/src/rty/projections.rs

```diff
@@ -48,22 +80,247 @@ fn without_constrs<T: TypeFoldable>(t: &T) -> T {
     t.fold_with(&mut WithoutConstrs)
 }
 
-struct WithPredicates {
-    proj_table: ProjectionTable,
+// -----------------------------------------------------------------------------------------------------
+// Code for normalizing `AliasTy` using impl -----------------------------------------------------------
+// -----------------------------------------------------------------------------------------------------
+
+fn into_rustc_generic_arg<'tcx>(
+    tcx: &TyCtxt<'tcx>,
+    bty: &GenericArg,
+) -> rustc_middle::ty::GenericArg<'tcx> {
+    match bty {
+        GenericArg::Ty(ty) => rustc_middle::ty::GenericArg::from(into_rustc_ty(tcx, ty)),
+        GenericArg::BaseTy(bty) => {
+            rustc_middle::ty::GenericArg::from(into_rustc_ty(tcx, &bty.clone().skip_binder()))
+        }
+        GenericArg::Lifetime(_) => todo!(),
+    }
 }
+fn into_rustc_bty<'tcx>(tcx: &TyCtxt<'tcx>, bty: &BaseTy) -> rustc_middle::ty::Ty<'tcx> {
+    match bty {
+        BaseTy::Int(i) => rustc_middle::ty::Ty::new_int(*tcx, *i), // rustc_middle::ty::Ty::mk_int(*tcx, int_ty),
+        BaseTy::Uint(i) => rustc_middle::ty::Ty::new_uint(*tcx, *i),
+        BaseTy::Param(pty) => pty.to_ty(*tcx),
+        BaseTy::Slice(ty) => rustc_middle::ty::Ty::new_slice(*tcx, into_rustc_ty(tcx, ty)),
+        BaseTy::Bool => rustc_middle::ty::Ty::new(*tcx, rustc_middle::ty::TyKind::Bool),
+        BaseTy::Char => rustc_middle::ty::Ty::new(*tcx, rustc_middle::ty::TyKind::Char),
+        BaseTy::Str => rustc_middle::ty::Ty::new_static_str(*tcx),
```

Comment:
This is wrong, we need [`tcx.types.str_`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.CommonTypes.html#structfield.str_)

Reply:
whoops. fixed here https://github.com/flux-rs/flux/pull/479